gear/gear/csrf.py

```diff
@@ -1,25 +1,27 @@
 import logging
 import secrets
-from functools import wraps
 
 from aiohttp import web
 
+from .auth import AIOHTTPHandler
+
 log = logging.getLogger('gear.auth')
 
 
 def new_csrf_token():
     return secrets.token_urlsafe(64)
 
 
-def check_csrf_token(fun):
-    @wraps(fun)
-    async def wrapped(request, *args, **kwargs):
+@web.middleware
+async def check_csrf_token(request: web.Request, handler: AIOHTTPHandler):
+    # CSRF prevention is only relevant to requests that use browser
+    # cookies for authentication.
+    if request.cookies and request.method in request.POST_METHODS:
```

Comment:
> request.POST_METHODS

`pyright` showed me this handy set but it doesn't seem to be documented. Perhaps we should just do `request.method == 'POST'`?

Reply:
let's list the "safe" methods instead.